File: java/base/src/main/java/vsdk/toolkit/raycasting/raytracing/SCREEN_ITERATE_CALLBACK.java
package vsdk.toolkit.raycasting.raytracing;

import vsdk.toolkit.common.color.ColorRgb;
import vsdk.toolkit.scene.Background;
import vsdk.toolkit.scene.Camera;
import vsdk.toolkit.scene.VoxelGrid;

@FunctionalInterface
public interface SCREEN_ITERATE_CALLBACK {
    ColorRgb call(Camera camera, VoxelGrid sceneVoxelGrid, Background sceneBackground, int x, int y, Object data);
}

File: typescript/base/src/main/vsdk/toolkit/raycasting/raytracing/SCREEN_ITERATE_CALLBACK.ts
import { ColorRgb } from "../../common/color/ColorRgb";
import { Background } from "../../scene/Background";
import { Camera } from "../../scene/Camera";
import { VoxelGrid } from "../../scene/VoxelGrid";

export type SCREEN_ITERATE_CALLBACK = (
  camera: Camera,
  sceneVoxelGrid: VoxelGrid,
  sceneBackground: Background | null,
  x: number,
  y: number,
  data: unknown
) => ColorRgb;
